### PR TITLE
Whyderrick

### PR DESCRIPTION
**Pull request by whyderrick**

### CODE REVIEW

File: teachable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Teachable
```
Comment (salogel42) [inline teachable.rb]: Why is this called teachable?  Maybe `HighFiveAble`?

File: apprentice_teacher.rb [hunk 2]
```diff
@@ -1,53 +1,23 @@
...
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
+    super
     @target_raise = 800
-    @phase = 3
-  end
-
-  def offer_high_five
-    "High five!"
+    @benchmark_rating = 80
```
Comment (salogel42) [inline apprentice_teacher.rb]: What are the tradeoffs between doing this as an instance variable vs constant? Why choose one over the other?

File: apprentice_teacher.rb [hunk 3]
```diff
@@ -1,53 +1,23 @@
...
 
-  def set_phase(num)
-    @phase = num
-    "Cool, I've always wanted to teach phase #{num}!"
+  def attend_training_session
+    puts "Whoa. I know ruby-fu"
   end
 
-  def teach_stuff
+    def teach_stuff
```
Comment (salogel42) [inline apprentice_teacher.rb]: Don't introduce bad indentation! Indentation may seem trivial, as the program works whether you're following standard conventions or not.  But when you're working on a large project with lots of other people, you want to know that everyone is following the same style guide.  Even just looking back on your own code, you'll prefer well formatted code over poorly formatted code!  And any workplace with code reviews in place will require that you follow their style guide, so get used to it now.

File: bonus.md [hunk 4]
```diff
@@ -4,10 +4,22 @@ Edit this file to answer the following questions. Give examples. You may need to
 
 1. Why is Hash.fetch useful?
 
+Hash fetch is useful because it allows us to set default values from our args hash without running into issues with booleans. Using the logical or can cause problems, especially when trying to set to boolean `false` or `nil`. Using `||`, those will never be true and thus, never be assigned. 
+
```
Comment (salogel42) [inline bonus.md:8]: Yup.  What happens when you use regular hash syntax and a key is not present? What does fetch do if it's not present (and no default is provided?)

File: bonus.md [hunk 5]
```diff
@@ -4,10 +4,22 @@ Edit this file to answer the following questions. Give examples. You may need to
...
 2. What is the disadvantage of single inheritance, and what's a possible solution to that problem?
 
+The disadvantage of single inheritance is that objects can only take bahovior from one other source. Given a we had a class of `Car` that drives, and a class of `Dog` that barks, we wouldn't be able to write very DRY code to build a thing that barks and drives, like a van that barks when the horn is pressed. 
+
+The most common solution for this is to prefer composition over inheritance and to extract behaviors into modules whenever possible.
```
Comment (salogel42) [inline bonus.md:17]: Good example!

File: bonus.md [hunk 6]
```diff
@@ -4,10 +4,22 @@ Edit this file to answer the following questions. Give examples. You may need to
...
 
+Classes are initialized with an options has to break the reliance on argument order. Since values are associate with key, we don't need to know that `@name` should be set to the first item in an array, and we can instead just grab whatever value matches the key of `"name"` or `:name`.
+
 4. What is the purpose of the private keyword in a class? What does it protect you from and why is that valuable?
 
+It makes methods that can only be called within the class. Usually it's used to separate helper methods and methods that update data. Using them this way makes it  much easier to ensure a careless method call doesn't upset the balance of your program. 
+
 5. Why are concepts like encapsulation, single responsibility, and abstraction important? Now that you've been programming for a while, have you seen any advantages for yourself, or can you imagine them in the future?
+
+These concepts help make programs more managable. They allow us to build a chain of behaviors so that we can identify where something went wrong more easily. If, instead of single-responsibility methods we had 1 long method with 10 responsibilities, it would be much harder to debug...
```
Comment (salogel42) [inline bonus.md:29]: Sounds good.

File: student.rb [hunk 7]
```diff
@@ -1,17 +1,16 @@
 class Student
+  include Teachable
+
   attr_reader :age, :phase
   attr_accessor :name
```
Comment (salogel42) [inline student.rb:18]: These look familiar... how can we refactor the similarities here vs `Teacher`?

**Review by salogel42 — COMMENTED** [review]
Make new commits addressing the questions and comments (feel free to ask clarifying questions if they don't make sense), and tag me again.

File: apprentice_teacher.rb (hunk 2 above)
Comment (whyderrick) [reply apprentice_teacher.rb]: This is answered in a comment on teacher now, but essentially it's a question that depends on how we evaluate teachers. If these are the same for every teacher at this level, then constants are the way to go. 

Honestly, I struggled with constants quite a lot on Monday night and wouldn't have used them in the timed test even if I'd realized they were the right choice for this. Got it now though 😄

File: apprentice_teacher.rb (hunk 3 above)
Comment (whyderrick) [reply apprentice_teacher.rb]: yup, sloppy accident.

File: student.rb (hunk 7 above)
Comment (whyderrick) [reply student.rb:18]: I wrote how I'd do this in comments, but I don't understand why this is a good refactor. I'd rather this 3 lines of repetitive code in two files than to create another dependency. I'd love to better understand though.

**Comment by whyderrick** [comment]
@salogel42 Refactored to answer your questions! looking forward to learning more on Monday!

**Pull request closed without merging**